modules/programs/sftpman.nix

```
@@ -0,0 +1,101 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.sftpman;
+
+  jsonFormat = pkgs.formats.json { };
+
+  mountOpts = { config, name, ... }: {
+    options = {
+      host = mkOption {
+        type = types.str;
+        description = "The host to connect to.";
+        default = null;
+      };
+      port = mkOption {
+        type = types.port;
+        description = "The port to connect to.";
+        default = 22;
+      };
+      user = mkOption {
+        type = types.str;
+        description = "The username to authenticate with.";
+        default = null;
+      };
+      mountOptions = mkOption {
+        type = types.listOf types.str;
+        description = "Options to pass to sshfs.";
+        default = [ ];
+      };
+      mountPoint = mkOption {
+        type = types.str;
+        description = "The remote path to mount.";
+        default = null;
+      };
+      authType = mkOption {
+        type = types.enum [
+          "password"
+          "publickey"
+          "hostbased"
+          "keyboard-interactive"
+          "gssapi-with-mic"
+        ];
+        description = "The authentication method to use.";
+        default = "publickey";
+      };
+      sshKey = mkOption {
+        type = types.nullOr types.str;
+        description = ''
+          Path to the ssh key to use for authentication.
+          Only applies if authMethod is `publickey`.
+        '';
+        default = cfg.defaultSshKey;
```

When an option default value depends on the configuration then you must provide a `defaultText` field to avoid making the documentation depend on the configuration. Something like this perhaps:

```suggestion
        default = cfg.defaultSshKey;
        defaultText = lib.literalExpression "config.programs.sftpman.defaultSshKey";
```